docs/education/dlt-fundamentals-course/lesson_2_dlt_sources_and_resources_create_first_dlt_pipeline.py

```diff
@@ -0,0 +1,719 @@
+    def all_data():
+      return my_df, get_genome_data, get_pokemon
+    ```
+
+    or for each resource separately:
+
+    ```python
+    @dlt.resource(table_name='pokemon_new', max_table_nesting=1)
+    def my_dict_list():
+        yield data
+    ```
+
+    In the example above, we want only 1 level of nested tables to be generated (so there are no nested tables of a nested table). Typical settings:
+
+    * `max_table_nesting=0` will not generate nested tables and will not flatten dicts into columns at all. All nested data will be represented as JSON.
+    * `max_table_nesting=1` will generate nested tables of root tables and nothing more. All nested data in nested tables will be represented as JSON.
+    """)
+    return
+
+
+@app.cell(hide_code=True)
+def _(mo):
+    mo.md(r"""
+    ---
+    ## **Exercise 1: Create a pipeline for GitHub API – repos endpoint**
+
+    In this exercise, you'll build a dlt pipeline to fetch data from the GitHub REST API. The goal is to learn how to use `dlt.pipeline`, `dlt.resource`, and `dlt.source` to extract and load data into a destination.
+
+    ## Instructions
+
+    1. **Explore the GitHub API**
+
+       Visit the [GitHub REST API Docs](https://docs.github.com/en/rest) to understand the endpoint to [list public repositories](https://docs.github.com/en/rest/repos/repos?apiVersion=2022-11-28) for an organization:
+
+       `GET https://api.github.com/orgs/{org}/repos`
+
+    2. **Build the pipeline**
+
+       Write a script to:
+
+       - Fetch repositories for the **dlt-hub** organization.
+       - Use `dlt.resource` to define the data extraction logic.
+       - Combine all resources into a single `@dlt.source`.
+       - Load the data into a DuckDB database.
+
+    3. **Inspect the data**
+
+       Use a `duckdb` connection, `sql_client`, or `pipeline.dataset()`.
+
+    > **Note**: For this exercise you don't need to use authentication or pagination.
+    """)
+    return
+
+
+@app.cell(hide_code=True)
+def _(mo):
+    mo.md(r"""Play with the API using the requests library:""")
+    return
+
+
+@app.cell
+def _(requests):
+    response = requests.get('https://api.github.com/orgs/dlt-hub/repos')
+    response.json()[0]
+    return
+
+
+@app.cell(hide_code=True)
+def _(mo):
+    mo.md(
+        r"""In the code snippet below, you will find an **example** for the **`events`** endpoint:"""
+    )
+    return
+
+
+@app.cell
+def _(DltResource, Iterable, TDataItems, dlt, requests):
+    @dlt.resource
+    def github_events() -> TDataItems:
+        url = 'https://api.github.com/orgs/dlt-hub/events'
+        response = requests.get(url)
+        yield response.json()
+
+    @dlt.source
+    def github_data() -> Iterable[DltResource]:
+        return (github_events,)
+    github_pipeline = dlt.pipeline(pipeline_name='github_pipeline', destination='duckdb', dataset_name='github_data')
+    _load_info = github_pipeline.run(github_data())
+    print(_load_info)
+    return
```

Comment:
We could include placeholders to make it more clear where the code for the exercise should go.
We had similar hints in the original notebooks, allthough not as explicit
```suggestion
@app.cell
def _(DltResource, Iterable, TDataItems, dlt, requests):
    @dlt.resource
    def github_events() -> TDataItems:
        url = 'https://api.github.com/orgs/dlt-hub/events'
        response = requests.get(url)
        yield response.json()

    # build the `github_repos` resource here

    @dlt.source
    def github_data() -> Iterable[DltResource]:
        return (github_events,)  # return your new resource as part of the source here
    github_pipeline = dlt.pipeline(pipeline_name='github_pipeline', destination='duckdb', dataset_name='github_data')
    _load_info = github_pipeline.run(github_data())
    print(_load_info)
    return
```